### PR TITLE
migrated to SCALE v2

### PR DESCRIPTION
**Pull request by cab404**
This is required for correct serialization in mvm pallet.

### CODE REVIEW

File: language/move-core/types/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,8 @@ bcs = { path = "../../../bcs", default-features = false }
 diem-crypto = { path = "../../../crypto/crypto", default-features = false }
 diem-crypto-derive = { path = "../../../crypto/crypto-derive", default-features = false }
 short-hex-str = { path = "../../../common/short-hex-str", default-features = false }
-parity-scale-codec = { version = "1.3.5", features = ["derive"], default-features = false }
+parity-scale-codec = { version = "2", features = ["derive"], default-features = false }
+parity-scale-codec-derive = { version = "2", default-features = false }
```
Comment (vitvakatu) [inline language/move-core/types/Cargo.toml:25]: Is it really needed here, as you're using `derive` feature?

File: mvm/Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,8 @@ hashbrown = "0.9"
 anyhow = { version = "1.0.34", default-features = false }
 hex = { version = "0.4.2", default-features = false, features = ["alloc"] }
 serde = { version = "1", default-features = false, package = "alt_serde", features = ["derive", "alloc"] }
-parity-scale-codec = { version = "1.3.5", default-features = false }
+parity-scale-codec = { version = "2", default-features = false }
```
Comment (vitvakatu) [inline mvm/Cargo.toml:15]: Do you need it here?

**Review by vitvakatu — COMMENTED** [review]
👍🏻

File: language/move-core/types/Cargo.toml (hunk 1 above)
Comment (cab404) [reply language/move-core/types/Cargo.toml:25]: Oops, should remove derive, as it's nonexistent as feature now.

**Comment by cab404** [comment]
This does fix serialization problems in pallet.

**Review by RIg410 — APPROVED** [review]

**Review by vitvakatu — APPROVED** [review]

**Review by fzzr- — APPROVED** [review]

**Pull request merged**